File: fintrack-backend/models/Transaction.js
const mongoose = require('mongoose');

const transactionSchema = new mongoose.Schema({
    userId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true,
        index: true
    },
    amount: {
        type: Number,
        required: true
    },
    type: {
        type: String,
        enum: ['INCOME', 'EXPENSE', 'DEBT'],
        required: true
    },
    categoryId: {
        type: String,
        required: true
    },
    categoryName: {
        type: String,
        required: false,
        default: 'Autre'
    },
    paymentMethod: {
        type: String,
        required: true
    },
    date: {
        type: Date,
        required: true
    },
    notes: {
        type: String,
        default: ''
    },
    localId: {
        type: String,
        index: true
    }
}, {
    timestamps: true
});

// Index pour recherche rapide par utilisateur et date
transactionSchema.index({ userId: 1, date: -1 });

module.exports = mongoose.model('Transaction', transactionSchema);

File: FinTrack3/app/src/main/java/bilal/com/fintrack/data/local/entities/Transaction.kt
package bilal.com.fintrack.data.local.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

enum class TransactionType {
    EXPENSE, INCOME, DEBT
}

@Entity(tableName = "transactions")
data class Transaction(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val amount: Double,
    val type: TransactionType,
    val categoryId: Long,
    val paymentMethod: String,
    val date: Long,
    val notes: String?,
    val userId: String = "",
    val createdAt: Long = System.currentTimeMillis(),
    val serverId: String? = null,  // MongoDB ObjectId for sync deduplication
    val categoryName: String? = null  // Store category name for reliable display
)
